src/main/room/main/G1HRPS/Room.java

```diff
@@ -1,33 +1,122 @@
 package main.G1HRPS;
+import java.util.ArrayList;
 
-import java.util.List;
-
+import Guest.Guest;
+/**
+ * This is a Room class object, containing details on a hotel room 
+ * @author LiangHee
+ *
+ */
 public class Room {
 
-	private int room_num_;
+	private String room_num_;
 	private RoomType room_type_;
 	private float room_price_;
 	private BedSize bedsize_;
 	private boolean wifi_enabled_;
 	private boolean w_view_;
 	private boolean w_smoking_;
 	private RoomStatus status_;
-	private List<Guest> guest_list_;
+	private ArrayList<Guest> guest_list_=new ArrayList<Guest>();
 
 	/**
-	 * 
-	 * @param room_type
-	 * @param price
-	 * @param room_number
-	 * @param bedSize
-	 * @param wifiEnabled
-	 * @param withView
-	 * @param Smoking
-	 * @param status
+	 * Creates a Room object to store information about the room.
+	 * @param room_type  The room types are Single, Standard, VIP, Suite, Deluxe.
+	 * @param price	 This is price of the room
+	 * @param room_number  This is room number of the room
+	 * @param bedSize	Bed size of bed in room
+	 * @param wifiEnabled Whether a room has wifi or not
+	 * @param withView	Whether a room has view or not
+	 * @param Smoking	Whether a room allows smoking or not
+	 * @param status	Possible room statuses are vacant, occupied, reserved, maintenance
 	 */
-	public Room(RoomType room_type, float price, int room_number, String bedSize, boolean wifiEnabled, boolean withView, boolean Smoking, RoomStatus status) {
-		// TODO - implement Room.Room
-		throw new UnsupportedOperationException();
+	public Room(RoomType room_type, float price, String room_number, BedSize bedSize, boolean wifiEnabled, boolean withView, boolean Smoking, RoomStatus status) {
+		room_num_=room_number;
+		room_type_=room_type;
+		room_price_=price;
+		bedsize_=bedSize;
+		wifi_enabled_=wifiEnabled;
+		w_view_=withView;
+		w_smoking_=Smoking;
+		status_=status;		
+	}
+	/**
+	 * Prints out general information of the room
+	 */
+	public void viewRoomDetails(){
+		System.out.println("room number: "+room_num_);
+		System.out.println("room type "+room_type_);
+		System.out.println("room price: "+room_price_);
+		System.out.println("bedsize: "+bedsize_);
+		System.out.println("wifi enabled: "+wifi_enabled_);
+		System.out.println("with view: "+w_view_);
+		System.out.println("with smoking: "+w_smoking_);
+		System.out.println("room status: "+status_);
+		System.out.println("guest staying: "+guest_list_);
+	}
+	/**
+	 * Prints out whether a room is available or not
+	 */
+	public void checkRoomAvailability() {
```

Comment:
Following the convention of all the other functions, the naming should be CheckRoomAvailability(), same for the rest.